### PR TITLE
DAOS-5339 tests: Enhance gathering dmg command output

### PR DESCRIPTION
**Pull request by phender**
Start migrating DmgCommand/DaosCommand helper methods to internally
process the output of the command and apply the regular expressions
in those methods in order to return a more user-friendly dictionary
of data.

Provide the base class changes to enable this functionality and
update DmgCommand.pool_create() as an example.

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/daos_utils.py [hunk 1]
```diff
@@ -21,12 +21,11 @@
   Any reproduction of computer software, computer software documentation, or
   portions thereof marked with this legend must also reproduce the markings.
 """
-from command_utils_base import \
-    CommandFailure, FormattedParameter, CommandWithParameters
-from command_utils import CommandWithSubCommand
+from daos_utils_base import DaosCommandBase
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/daos_utils.py:24]: Added missing new file

File: src/tests/ftest/util/command_utils.py [hunk 2]
```diff
@@ -467,25 +479,64 @@ def set_sub_command(self, value):
...
             raise CommandFailure(
                 "<{}> command failed: {}".format(self.command, error))
+        return self.result
+
+    def _get_result(self, sub_command_list=None, **kwargs):
+        """Get the result from running the command with the defined arguments.
+
+        The optional sub_command_list and kwargs are used to define the command
+        that will be executed.  If they are excluded, the commnad will be run as
+        it currently defined.
```
Comment (shimizukko) [inline src/tests/ftest/util/command_utils.py:507]:
```suggestion
        that will be executed.  If they are excluded, the command will be run as defined.
```
or something similar.

**Comment by amjustin13** [comment]
Just a suggestion on how the the helper functions on `daos_utils.py` and `dmg_utils.py` are setup. I think that it would be more readable to have the style format of the function similar to the `dmg_utils.storage_prepare()` function:
```
kwargs = {
            "arg1": arg1,
            "arg2": arg2,
            "arg3": arg3,
}
return self._get_result(("storage", "prepare"), **kwargs)
```
or maybe something like:
```
return self._get_result(("storage", "prepare"),
                         "arg1"=arg1,
                         "arg2"=arg2,
                         "arg3"=arg3)
```

File: src/tests/ftest/util/command_utils.py (hunk 2 above)
Comment (phender) [reply src/tests/ftest/util/command_utils.py:507]: Thanks, I like that better.  I'll apply the change in a follow-on PR.

**Comment by phender** [comment]
The single failure is an known issue: https://jira.hpdd.intel.com/browse/DAOS-5321

**Comment by phender** [comment]
> return self._get_result(("storage", "prepare"),
>                          "arg1"=arg1,
>                          "arg2"=arg2,
>                          "arg3"=arg3)

I was somewhat intentionally trying to keep the line number count down, but in cases where multiple lines are needed I could see using a format similar the following being useful:
```
return self._get_result(
    ("storage", "prepare"),
    "arg1"=arg1, "arg2"=arg2, "arg3"=arg3)
```

**Pull request merged**